src/main/java/seedu/pill/command/DeleteItemCommand.java

```diff
@@ -1,19 +1,31 @@
 package seedu.pill.command;
 
 import seedu.pill.exceptions.PillException;
+import seedu.pill.util.Item;
 import seedu.pill.util.ItemMap;
 import seedu.pill.util.Storage;
 
+import java.time.LocalDate;
+
 public class DeleteItemCommand extends Command {
     private final String itemName;
+    private final LocalDate expiryDate;
 
     public DeleteItemCommand(String itemName) {
         this.itemName = itemName;
+        this.expiryDate = null;
+    }
+
+    public DeleteItemCommand(String itemName, LocalDate expiryDate) {
+        this.itemName = itemName;
+        this.expiryDate = expiryDate;
     }
```

Comment:
I see you are indexing them based on expiry dates. Might be a problem if there are items that are not based on expiry dates but that can be fixed down the line once we firm up the data structure.